stout/flags.cc

```diff
@@ -133,7 +133,76 @@ void Parser::Parse(int* argc, const char*** argv) {
 ////////////////////////////////////////////////////////////////////////
 
 void Parser::Parse(
-    const std::multimap<std::string, std::optional<std::string>>& values) {
+    std::multimap<std::string, std::optional<std::string>>& values) {
+  // Parse environment variables if environment_variable_prefix_
+  // contains the specific prefix of variables we want to parse.
+  if (environment_variable_prefix_) {
+    // Get the length of the included prefix.
+    const auto prefix_length = environment_variable_prefix_.value().length();
+
+    // Run through all environment variables and select those which has
+    // prefix (environment_variable_prefix_) in the name. Then we grab
+    // correct name and the value and store this pair in `pairs_to_parse`.
+    // We use global variable `environ` to have access to the environment
+    // variables to be able to select all necessary variables for
+    // parsing according to the included prefix (check class `Parser`
+    // in the `stout/flags.h` which has `environment_variable_prefix_`
+    // variable of the type `std::optional<std::string>`).
+    for (char** env = environ; *env != nullptr; ++env) {
+      const char* c_name_value =
+          strstr(*env, environment_variable_prefix_.value().c_str());
+      if (!c_name_value) {
+        continue;
+      }
+
+      std::string name_value{c_name_value};
+
+      size_t end_name = name_value.find("=");
+      if (end_name == std::string::npos) {
+        continue;
+      } else {
+        std::string name;
+        std::optional<std::string> value;
+
+        // By default all environment variables are as follows:
+        //    name=value
+        // That's why we need to grab correctly the part with name
+        // and the part with value.
+
+        // Skip first symbol of the name if it is not a number or
+        // a letter.
+        size_t skip = 0;
+        if (const char& symbol = name_value.at(prefix_length);
+            !std::isdigit(symbol) && !std::isalpha(symbol)) {
+          skip += 1;
+        }
+
+        // Grabbing the part with name.
+        name = name_value.substr(
+            prefix_length + skip,
+            end_name - prefix_length - skip);
+
+        // It's possible that users can set variables with upper cases.
+        // So we should be sure that names we pass for parsing have
+        // only lower-cases symbols.
+        std::transform(
+            name.begin(),
+            name.end(),
+            name.begin(),
+            [](auto symbol) { return std::tolower(symbol); });
+
+        // Grab the part with the value.
+        if (end_name != name_value.size() - 1) {
+          // Do we suppose that the values from the environment variables
+          // should be always strings?
+          value = name_value.substr(end_name + 1, std::string::npos);
+        }
+
+        values.emplace(name, value);
```

Comment:
I'd like to use more of `absl` here, for example:

```suggestion
      std::vector<std::string> name_value = absl::StrSplit(*env, absl::MaxSplits('=', 1));

      CHECK_EQ(name_value.size(), 2)
          << "Expecting all environment variables to have '=' delimiter";

      std::string name = name_value[0];
      std::string value = name_value[1];

      if (absl::ConsumePrefix(&name, *environment_variable_prefix_)) {
        // Normalize all names to lowercase.
        absl::AsciiStrToLower(&name);
        values.emplace(name, value);
      }
```

Reply:
ok